src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -312,10 +302,6 @@ private void OnWriteCompleted(WriteContext writeContext)
                 {
                     scheduleWrite = true;
```

Comment:
Hmm... maybe. 

I also think the `WriteContext` and `SocketOutput` can be merged in someway, as essentially its only unique state is `SocketShutdownSend`+`SocketDisconnect`; but if those are use/triggered when all the blocks are processed rather than depending on the active `WriteContext` they can be moved to properties of `SocketOutput`?